lib/ansible/modules/network/ios/ios_ntp.py

```diff
@@ -0,0 +1,321 @@
+        source_int_have = None
+
+    try:
+        source_int_have = have[0]['source_int']
+    except KeyError:
+        source_int_have = None
+
+    try:
+        acl_have = have[0]['acl']
+    except KeyError:
+        acl_have = None
+
+    try:
+        logging_have = have[0]['logging']
+    except KeyError:
+        logging_have = None
+
+    try:
+        auth_have = have[0]['auth']
+    except KeyError:
+        auth_have = None
+
+    try:
+        auth_key_have = have[0]['auth_key']
+    except KeyError:
+        auth_key_have = None
+
+    try:
+        key_id_have = have[0]['key_id']
+    except KeyError:
+        key_id_have = None
+
+    for w in want:
+        server = w['server']
+        source_int = w['source_int']
+        acl = w['acl']
+        logging = w['logging']
+        state = w['state']
+        auth = w['auth']
+        auth_key = w['auth_key']
+        key_id = w['key_id']
+
+        if state == 'absent':
+            if server_have and server in server_have:
+                commands.append('no ntp server {0}'.format(server))
+            if source_int and source_int_have:
+                commands.append('no ntp source {0}'.format(source_int))
+            if acl and acl_have:
+                commands.append('no ntp access-group peer {0}'.format(acl))
+            if logging is True and logging_have:
+                commands.append('no ntp logging')
+            if auth is True and auth_have:
+                commands.append('no ntp authenticate')
+            if key_id and key_id_have:
+                commands.append('no ntp trusted-key {0}'.format(key_id))
+            if auth_key and auth_key_have:
+                commands.append('no ntp authentication-key {0} md5 {1} 7'.format(key_id, auth_key))
+
+        elif state == 'present':
+            if server is not None and server not in server_have:
+                commands.append('ntp server {0}'.format(server))
+            if source_int is not None and source_int != source_int_have:
+                commands.append('ntp source {0}'.format(source_int))
+            if acl is not None and acl != acl_have:
+                commands.append('ntp access-group peer {0}'.format(acl))
+            if logging is not None and logging != logging_have and logging is not False:
+                commands.append('ntp logging')
+            if auth is not None and auth != auth_have and auth is not False:
+                commands.append('ntp authenticate')
+            if key_id is not None and key_id != key_id_have:
+                commands.append('ntp trusted-key {0}'.format(key_id))
+            if auth_key is not None and auth_key != auth_key_have:
+                commands.append('ntp authentication-key {0} md5 {1} 7'.format(key_id, auth_key))
+
+    return commands
+
+
+def main():
+
+    argument_spec = dict(
+        server=dict(),
+        source_int=dict(),
+        acl=dict(),
+        logging=dict(type='bool', choices=[True, False], default=False),
```

Comment:
Also remove `choices` for these, `type='bool'` handles that as well

Reply:
Done